File: tests/fixtures/_highlight-corpus/sample.fs
// F# sample
module Sample

open System.IO

type Counter = { Value: int } with
    member this.Increment n = { Value = this.Value + n }

let countBytes (path: string) : int =
    if not (File.Exists path) then
        eprintfn "input not found: %s" path
        0
    else
        File.ReadLines(path)
        |> Seq.filter (fun line -> line.Length > 0)
        |> Seq.fold (fun (c: Counter) (line: string) -> c.Increment line.Length) { Value = 0 }
        |> fun c -> c.Value

[<EntryPoint>]
let main argv =
    let path = if argv.Length > 0 then argv.[0] else "input.txt"
    let total = countBytes path
    printfn "total bytes: %d" total
    0

File: tests/fixtures/_highlight-corpus/sample.r
# R sample
read_lines <- function(path) {
  con <- file(path, "r")
  on.exit(close(con))
  readLines(con)
}

count_bytes <- function(lines) {
  total <- 0L
  for (line in lines) {
    if (nchar(line) > 0L) {
      total <- total + nchar(line)
    }
  }
  total
}

main <- function(argv) {
  path <- if (length(argv) > 0) argv[1] else "input.txt"
  if (!file.exists(path)) {
    message("input not found: ", path)
    return(invisible(1L))
  }
  lines <- read_lines(path)
  total <- count_bytes(lines)
  cat(sprintf("total bytes: %d\n", total))
  invisible(0L)
}

if (!interactive()) {
  args <- commandArgs(trailingOnly = TRUE)
  status <- main(args)
  quit(status = status)
}
